src/Catalog.Service/Infrastructure.Messages/DependencyInjection.cs

```diff
@@ -0,0 +1,32 @@
+﻿using ApplicationCore.Common.Interfaces;
+using Azure.Messaging.ServiceBus;
+using Infrastructure.Messages.Options;
+using Infrastructure.Messages.Services;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Infrastructure.Messages;
+
+public static class DependencyInjection
+{
+    public static IServiceCollection ConfigureInfrastructureMessageBusServices(
+        this IServiceCollection services,
+        Action configureMessageBusOptions)
+    {
+        configureMessageBusOptions();
+
+        services.AddAzureClientsCore(false);
+        services.AddSingleton(sp =>

```

Comment:
I'd suggest using Azure SDK for registration, see:
[AddAzureClients](https://learn.microsoft.com/en-us/dotnet/api/overview/azure/messaging.servicebus-readme?view=azure-dotnet#registering-with-aspnet-core-dependency-injection)

Reply:
Actually i did because AddAzureclients does not have IServiceProvider, so Options pattern is impossible with it.
updated to init options via Bind